wfe2/wfe.go

```diff
@@ -2278,6 +2278,37 @@ func (wfe *WebFrontEndImpl) RenewalInfo(ctx context.Context, logEvent *web.Reque
 	logEvent.Extra["RequestedSerial"] = serial
 	beeline.AddFieldToTrace(ctx, "request.serial", serial)
 
+	// Check if the serial is part of an ongoing incident.
+	result, err := wfe.sa.IncidentsForSerial(ctx, &sapb.Serial{Serial: serial})
+	if err != nil && !errors.Is(err, berrors.NotFound) {
+		wfe.sendError(response, logEvent, probs.ServerInternal("Unable to check if the serial is impacted by an incident"), err)
```

Comment:
I think it would be reasonable to just continue onwards if this fails. Then the client would at least get _some_ suggested renewal window, even if it's not the "renew right now" window that we'd like them to get.

But that might make this code kinda non-linear, and papering over 500s isn't usually the best idea. Up to you.

Reply:
The way I see it, there are two situations that could cause this `500`:

Situation 1: We return a 500 because of an issue between WFE ⭢ SA or SA ⭢ DB. In this situation we would return a similar 500 but only after incurring a (failed) call to `sa.GetCertificate()` in the next block.

Situation 2: We return a 500 because we're missing the `incidents` table in Production. This would result in a mysql.Error(1146) "Table doesn't exist". SRE has an issue to ensure that the `incidents` table is provisioned ahead of this change.

I'm not for papering over a legitimate 500 especially when we're likely to encounter the same error inside of the next block.